packages/back-end/src/services/experiments.ts

```diff
@@ -933,15 +933,11 @@ export function toSnapshotApiInterface(
   };
 }
 
-/**
- * While the `postMetricValidator` can detect the presence of values, it cannot figure out the correctness.
- * @param payload
- * @param datasource
- */
-export function postMetricApiPayloadIsValid(
-  payload: z.infer<typeof postMetricValidator.bodySchema>,
-  datasource: Pick<DataSourceInterface, "type">
-): { valid: true } | { valid: false; error: string } {
+const postOrPutMetricPayloadIsValid = (
+  payload:
+    | z.infer<typeof postMetricValidator.bodySchema>
+    | z.infer<typeof putMetricValidator.bodySchema>
+): { valid: true } | { valid: false; error: string } => {
```

Comment:
This doesn't seem like the right abstraction. Any changes you make here can impact the post endpoint. I think it makes sense to have a separate validator and not create this abstraction in an attempt to DRY up the code.

I see the `postMetricApiPayloadIsValid` is defined below—did you attempt to do a 2-in-1 validator here, realize it didn't work, and forget to restore the name? Or is the other one unused?